### DISCUSSION TITLE
Error during migration using web3modal v1 and ethereum-provider v2

### DISCUSSION THREAD

**Discussion by d-hrituc** [WalletConnect/walletconnect-monorepo#2968] [Developer Support]
Hi everyone, I'm working on migrating to the new v2 version using the least invasive migration technique listed on the guide. It is working fine on desktop browser but I get quite a cryptic error when trying it on a mobile browser, pointing to some line in the web3modal dist folder (see below).
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/94625227/c1822b7a-7284-4ce5-9c2b-a01874287093)
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/94625227/631a914e-5fc7-4057-bd08-083a0ed8dbe9)

This are the related packages that I have installed:

```
"@walletconnect/ethereum-provider": "^2.8.4",
"@walletconnect/modal": "^2.5.4",
"web3modal": "^1.9.8",
"typescript": "^5.1.3"
```

This is how I initiate the provider:

```
async function initProvider() {
    const provider = await EthereumProvider.init({
        projectId: '{PROJECT_ID}',
        chains: [80001],
        showQrModal: true,
    });
    return provider;
}
```

and how I pass it to the constructor of the web3modal:

```
window.repo = new Web3Modal({
    network: "mumbai",
    cacheProvider: true,
    providerOptions: {
        walletconnect: {
            package: provider,
            options: {
                rpc: {
                    80001: "https://polygon-mumbai.g.alchemy.com/v2/{KEY}",
                },
            },
        },
    },
});
```

Is there anything thay I'm missing? Any help is appreciated. 

Thank you

**Comment by boidushya** [comment]
Hey! Would you be able to provide a reproducible example of this error for us to test? A [codesandbox](https://csb.app) would be perfect for this situation.

**Reply by d-hrituc** [reply]
Hi, yes, I could set it up. Should I leave the `KEY` and `PROJECT_ID` placeholders or are there any test projects that I can use?

**Reply by boidushya** [reply]
Yea sure, keeping them as placeholder works

**Reply by d-hrituc** [reply]
Awesome, here is the [codesandbox](https://codesandbox.io/s/cocky-dirac-n3vf96?file=/src/index.js)

**Reply by boidushya** [reply]
Looks like this is a very early version of web3modal youre using. Can you try upgrading to latest? Heres the [documentation](https://docs.walletconnect.com/2.0/web/web3modal/html/wagmi/installation) for it

**Reply by d-hrituc** [reply]
Well, yeah, it is an older version, but this [migration guide](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#web3modal-v10) states that Web3Modal v1.0 can still be used by just upgrading the `ethereum-provider`

**Reply by boidushya** [reply]
Hey, I dont see this in our docs. The existing [Web3Modal v1 migration steps](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#web3modal-v10) instructs to either switch to web3modal v2 or WalletConnectModal.
<img width="875" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/26746725/e33503bb-1f73-4e77-b0b9-8bccd9b4ecf0">

**Reply by d-hrituc** [reply]
Oh well, that part just got edited out, because just a few hours ago it was like this:
![MicrosoftTeams-image (2)](https://github.com/WalletConnect/walletconnect-monorepo/assets/94625227/f1e36574-c821-4de7-8a48-20ef0768c10e)

Edit: now I see that it was updated June 30

**Reply by boidushya** [reply]
All good, must have been cached by your browser. Hope updating the modal to v2 resolves this issue for you. If it doesnt, feel free to reopen this thread
